File: api/src/main/java/com/mindcare/api/model/UserProfessional.java
package com.mindcare.api.model;

import com.mindcare.api.utils.LanguagesConverter;

import lombok.Data;
import lombok.EqualsAndHashCode;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Column;
import jakarta.persistence.Convert;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.JoinTable;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.FetchType;
import java.util.Set;
import java.util.HashSet;
import java.util.List;

@Entity
@Table(name = "user_professional")
@Data
@EqualsAndHashCode(callSuper = true)
public class UserProfessional extends User {

  @Column(nullable = false)
  @Convert(converter = LanguagesConverter.class)
  private List<String> languages;

  @Column(nullable = false)
  private Integer experience;

  @Column(nullable = false)
  private String certification;

  @Column(nullable = false)
  private String companyName;

  @Column(nullable = false, unique = true)
  private String medicalIdentificationNumber;

  @Column(nullable = false, unique = true)
  private String companyIdentificationNumber;

  @ManyToMany(fetch = FetchType.EAGER)
  @JoinTable(
    name = "professional_therapy_domain",
    joinColumns = @JoinColumn(name = "professional_uuid"),
    inverseJoinColumns = @JoinColumn(name = "therapy_domain_id")
  )
  private Set<TherapyDomain> therapyDomains = new HashSet<>();
}

File: api/app/Models/UserProfessional.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class UserProfessional extends Model
{
    use HasUuids;

    protected $table = 'user_professionals';

    protected $primaryKey = 'uuid';

    public $timestamps = false;

    protected $fillable = [
        'uuid',
        'languages',
        'experience',
        'certification',
        'company_name',
        'medical_identification_number',
        'company_identification_number',
    ];

    protected $casts = [
        'languages' => 'array',
        'experience' => 'integer',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'uuid', 'uuid');
    }

    public function therapyDomains(): BelongsToMany
    {
        return $this->belongsToMany(TherapyDomain::class, 'professional_therapy_domain', 'professional_uuid', 'therapy_domain_id');
    }
}
